From 79edd84b50a170cb96a30e4efc1d719663cbfaf0 Mon Sep 17 00:00:00 2001
From: Julien Thomas <jthomas@zenetys.com>
Date: Wed, 10 Aug 2022 17:39:58 +0200
Subject: [PATCH] conf: Fix #pragma GCC diagnostic not allowed inside functions

Commit 829a6f6 "CI: improve use of github action matrix feature;
optimization" has introduced use of #pragma GCC diagnostic in function
doNameLine() of runtime/conf.c. This is not supported by old compilers
like GCC 4.4.7 on el6.
---
 runtime/conf.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/runtime/conf.c b/runtime/conf.c
index 3e56301ef..4c13e76c5 100644
--- a/runtime/conf.c
+++ b/runtime/conf.c
@@ -165,13 +165,17 @@ doNameLine(uchar **pp, void* pVal)
 	p = *pp;
 	assert(p != NULL);
 
+#if defined(__GNUC__) && __GNUC__ >= 5
 	PRAGMA_DIAGNOSTIC_PUSH
 	PRAGMA_IGNORE_Wvoid_pointer_to_enum_cast;
+#endif
 	/* this time, pVal actually is NOT a pointer! It is save to case, as
 	 * the enum was written to it, so there can be no loss of bits (ptr is larger).
 	 */
 	eDir = (enum eDirective) pVal;
+#if defined(__GNUC__) && __GNUC__ >= 5
 	PRAGMA_DIAGNOSTIC_POP
+#endif
 
 	if(getSubString(&p, szName, sizeof(szName), ',')  != 0) {
 		LogError(0, RS_RET_NOT_FOUND, "Invalid config line: could not extract name - line ignored");
-- 
2.21.1

